### PR TITLE
Fix SpinBox will reset unsubmitted text when redrawing

### LINKED ISSUE #79507: SpinBox resets text when tooltip is showed
### Godot version

4.1

### System information

win 10

### Issue description

If you type text (not pressing enter) and hover mouse so tooltip shows up, then text dissapears.

bug

### Steps to reproduce

1. Add a SpinBox
2. Add a Tooltip Text
3. Run
4. type some text. DON'T PRESS ENTER
5. wait a second until tooltip will show up
6. Observe how text dissapeared.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by jsjtxietian**
This should fix #79507

As I mentioned in the issue, this [commit](https://github.com/godotengine/godot/commit/9500f8e69ae798c070c4daca9c46beaf8db18bd4) added `_update_text()` to `NOTIFICATION_DRAW`. And insided `_update_text()`, it uses the shared range value to update the `line_edit`'s text, but the value will override the unsubmited user's input. 

Originally this commit makes sense because user may call `set_value_no_signal` which will directly change shared range's value, but when a user input some text and then triggered `NOTIFICATION_DRAW`, this could be wrong and cause this bug.

So we will encounter a situation where we have two sources of value that need to consider in `_update_text`, one from script like `set_value_no_signal` which directly change range's value , one from user's uncommited input reflected in `line_edit` compoment. 

Basically what my fix do is that it trys to remember the last_updated_text, and when the value get from shared range equals this last_updated_text but not euqal to line_edit's current value, just return to keep line_edit's current value. That being saied, when a user is inputing something and some script called `set_value_no_signal` at the same time, this code will keep the shared range's value instead of user's input.

It should work in most cases, I know this is not the best fix, but I think it is less invasive in this way. I would appreciate any feedback or alternative suggestions.

### CODE REVIEW

**Comment by KoBeWi** [comment]
When you press Escape to unfocus the SpinBox, anything you wrote will remain.

**Comment by jsjtxietian** [comment]
> When you press Escape to unfocus the SpinBox, anything you wrote will remain.

I guess this is not the expected behaviour? If the user do not turn on autosave, I guess it should drop the content after unfocus?

**Comment by AThousandShips** [comment]
I'd say pressing escape should cancel any input

**Comment by KoBeWi** [comment]
Yes, it's a bug. You can write some invalid non-numeric value and it will linger.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
